### PR TITLE
Spawn INDEXERs lazily from partition

### PR DESCRIPTION
**Pull request by lava**
Instead of going over the whole partition and spawning all indexer when loading a partition, we `mmap` the partition and only access the relevant part of the memory once we get a query that actually uses this indexer.

### CODE REVIEW

File: libvast/vast/system/partition.hpp [hunk 1]
```diff
@@ -14,6 +14,7 @@
 #pragma once
 
 #include "vast/aliases.hpp"
+#include "vast/chunk.hpp"
```
Comment (dominiklohmann) [inline libvast/vast/system/partition.hpp:17]: We should only need a forward-declaration of `chunk_ptr` here, not the full header, so `vast/fwd.hpp` should suffice, and if it does not, this should be solved in `vast/fwd.hpp` instead.

The `vast/detail/stable_map.hpp` include can be removed I think.

**Review by dominiklohmann — APPROVED** [review]

File: libvast/src/system/partition.cpp [hunk 2]
```diff
@@ -551,7 +554,7 @@ passive_partition(caf::stateful_actor<passive_partition_state>* self, uuid id,
   // to queries. The `skip` default handler is used to buffer all messages
   // arriving until then.
   self->set_default_handler(skip);
-  self->send(caf::actor_cast<caf::actor>(fs), atom::read_v, path);
+  self->send(caf::actor_cast<caf::actor>(fs), atom::mmap_v, path);
```
Comment (mavam) [inline libvast/src/system/partition.cpp:557]: It's great to see that we literally only need a one-line change to switch to mmap'ing now!

File: libvast/src/system/partition.cpp [hunk 3]
```diff
@@ -65,19 +65,39 @@ using namespace caf;
...
-namespace {
-
-// The functions in this namespace take PartitionState as template argument
-// because the impelementation is the same for passive and active partitions.
+caf::actor active_partition_state::indexer_at(size_t position) const {
+  VAST_ASSERT(position < indexers.size());
+  return as_vector(indexers)[position].second;
+}
 
 /// Gets the INDEXER at a certain position in the `indexers` stable map.
```
Comment (mavam) [inline libvast/src/system/partition.cpp:73]: I think the comment no longer holds, right?

**Review by mavam — APPROVED** [review]
Nice work!

I just noticed one detail in a comment.

File: libvast/src/system/partition.cpp (hunk 3 above)
Comment (lava) [reply libvast/src/system/partition.cpp:73]: Right, updated.

**Pull request merged**